src/components/BatchSearchForm.vue

```diff
@@ -317,102 +370,156 @@ export default {
     advancedFiltersIcon() {
       return this.showAdvancedFilters ? 'angle-down' : 'angle-right'
     },
-    fuse() {
+    fuseFileTypes() {
       const keys = ['extensions', 'label', 'mime']
       const options = {
         distance: 100,
         keys,
         shouldSort: true
       }
       return new Fuse(this.allFileTypes, options)
+    },
+    fuseTags() {
+      const options = {
+        distance: 100,
+        shouldSort: true
+      }
+      return new Fuse(this.allTags, options)
     }
   },
   watch: {
     phraseMatch() {
-      this.$set(this, 'fuzziness', 0)
+      this.selectedFuzziness = 0
     },
-    projects() {
-      this.$set(this, 'fileType', '')
-      this.$set(this, 'fileTypes', [])
-      this.$set(this, 'paths', [])
-      this.$set(this, 'allFileTypes', [])
+    async projects() {
+      this.resetProjectData()
       this.hideSuggestionsFileTypes()
-      this.retrieveFileTypes()
+      this.hideSuggestionsTags()
+      await Promise.all([this.retrieveFileTypes(), this.retrieveTags()])
     },
-    showAdvancedFilters() {
-      this.retrieveFileTypes()
+    async showAdvancedFilters() {
+      await Promise.all([this.retrieveFileTypes(), this.retrieveTags()])
     }
   },
   created() {
-    this.$set(this, 'selectedProjects', this.defaultSelectedProjects)
+    this.selectedProjects = this.defaultSelectedProjects
   },
   methods: {
+    createQueryTemplate() {
```

Comment:
This can be transformed as a computed prop
We can keep it as a **method** with this signature:
```js
createQueryTemplate(filters){
  const textFilters = filters.map(f=>{
    const textFilter = new FilterText({ name: f.name, key: f.key, forceExclude: f.forceExclude })
    textFilter.values = f.values
    return textFilter
  })

  const { query } = this.$core.api.elasticsearch
    .rootSearch(textFilters, TEMPLATE_VALUE.QUERY)
    .build()
  return JSON.stringify(query)
}
```
and have a **computed** prop 
```js
queryTemplate(){
  const filters = [
    { name: 'tags', key: 'tags', forceExclude: this.forceExclude, values: this.tags},
    { name: 'contentType', key: 'contentType' , forceExclude: false, values: this.fileTypes },
    {name: 'path', key: 'byDirname', forceExclude: false, values: this.paths }
  ]

  return createQueryTemplate(filters)
}
```
This will be extensible if we want to add exclude filters on other attributes

Reply:
Good way to refactor. The only catch is that `path`is a `FilterPath` type instead of `FilterText`

Reply:
Oups therefore it's possible to add switch case in the map  :

```js
...
const textFilters = filters.map(f=>{ 
  let filter;
  switch (f.name) {
   case 'path':
    filter = new FilterPath({ name: f.name, key: f.key, forceExclude: f.forceExclude }) 
    break;
  case 'contentType':
  case 'tags:
    filter = new FilterText({ name: f.name, key: f.key, forceExclude: f.forceExclude })
    break;
  default: 
    throw new Error("unsupported query filter")
  }
  filter.values = f.values 
  return filter
 })
...